kep/260-network-aware-scheduling/README.md

```diff
@@ -0,0 +1,1398 @@
+                              x-kubernetes-int-or-string: true
+                            maxNetworkCost:
+                              type: integer
+                              default: 1
+                              minimum: 0
+                              maximum: 100
+                              format: int64
+                              description: The maximum network cost between both Pods.
+                            required:
+                              - podName
+          status:
+            description: Record Pod Allocations (Pod id, hostnames).
+            properties:
+              runningPods:
+                description: The number of actively running pods.
+                format: int32
+                type: integer
+                minimum: 0
+              scheduled:
+                type: array
+                additionalProperties:
+                  podName:
+                    type: string
+                    description: Name of the Pod.
+                  replicaID:
+                    type: string
+                    description: Pod ID
+                  hostname:
+                    type: string
+                    description: Hostname allocating Pod instance
+              scheduleStartTime:
+                description: ScheduleStartTime of the AppGroup
+                format: date-time
+                type: string
+              topologyCalculationTime:
+                description: topologyCalculationTime of the AppGroup
+                format: date-time
+                type: string
+              topologyOrder:
+                description: The optimal order to schedule pods on this App Group based on a given algorithm.
+                type: array
+                items:
+                  description: Pod Name and Pod Priority index
+                  type: object
+                    podName:
+                      type: string
+                      description: Name of the Pod.
+                    index:
+                      type: integer
+                      default: 1
+                      minimum: 1
+                      format: int64
+                      description: Priority index for each Pod (e.g., 1, 2, ...)
+                        (1 means pod should be scheduled first in the AppGroup)
+            type: object
+        type: object
+    served: true
+    storage: true
+status:
+  acceptedNames:
+    kind: ""
+    plural: ""
+  conditions: []
+  storedVersions: []
+```
+
+### Example
+
+<p align="center"><img src="figs/chain.png" title="Chain" width="600" class="center"/></p>
+
+```yaml
+# Example App Group CRD spec
+apiVersion: scheduling.sigs.k8s.io/v1alpha1
+kind: AppGroup
+metadata:
+  name: A1
+spec:
+  numMembers: 3
+  topologySortingAlgorithm: KahnSort
+  Pods:
+    - podName: P1
+      dependencies:
```

Comment:
Is `dependencies` trying to abstract a DAG among the pods? If so, it's another topic, and I don't think we should include in this KEP.

Reply:
@Huang-Wei , again, if we remove the DAG part, we will not have latency/network cost requirements for microservices in the application. Then, the only thing we add is to patch node bandwidth capacity as an extended resource. In that case, we do not need new scheduler plugins and can reuse existing ones for extended resources.